### PR TITLE
Added CI

### PR DESCRIPTION
**Pull request by AviiNL**
Automatically build Linux and Windows versions of Graphite on changes in the master branch (which includes accepting/merging pull requests)

Releases are a zip file for windows containing the executable and opencv_world library. and a AppImage for linux.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,133 @@
+name: linux_x64
```
Comment (flibidydibidy) [inline .github/workflows/build.yml]: This should probably be 'graphite_build' or 'graphite_ci' or something like that.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+          msbuild ALL_BUILD.vcxproj /p:Configuration=Release
+
+      - name: zip the graphite binary
+        if: success()
+        run: |
+          mkdir $env:GITHUB_WORKSPACE/release
+          cd $env:GITHUB_WORKSPACE/release
+
+          cp $env:GITHUB_WORKSPACE/build/graphite/Release/*.exe .
+          cp $env:GITHUB_WORKSPACE/3rd/opencv/build/x64/vc16/bin/opencv_world480.dll .
```
Comment (flibidydibidy) [inline .github/workflows/build.yml:96]: do we also need 'opencv_videoio_ffmpeg480'? and SDL2.dll?

File: .gitmodules [hunk 3]
```diff
@@ -15,4 +15,4 @@
 	url = https://github.com/gabime/spdlog.git
 [submodule "3rd/json"]
 	path = 3rd/json
-	url = https://github.com/nlohmann/json.git
+	url = https://github.com/nlohmann/json.git
```
Comment (flibidydibidy) [inline .gitmodules:18]: is it possible to not have this file show up as changed at all?

File: README.md [hunk 4]
```diff
@@ -1,35 +1,35 @@
...
------
 #### One time setup
-Install dependencies 
+
+Install dependencies
+
 ```
 sudo apt install build-essential cmake git ninja-build \
     libavcodec-dev libavformat-dev libswscale-dev libv4l-dev \
     mesa-common-dev xorg-dev
```
Comment (flibidydibidy) [inline README.md]: we should probably change this to align with build.yml.

File: README.md [hunk 5]
```diff
@@ -96,14 +97,13 @@ mkdir build32
 cd build64
```
Comment (flibidydibidy) [inline README.md]: would you mind fixing this typo? cd build64 > cd build32

**Review by flibidydibidy — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (AviiNL) [reply .github/workflows/build.yml]: Oh yeah, i initially intended to split it up in multiple files

File: .github/workflows/build.yml (hunk 2 above)
Comment (AviiNL) [reply .github/workflows/build.yml:96]: it works for me without them, will need testing from the community if they're reporting missing libs

File: .gitmodules (hunk 3 above)
Comment (AviiNL) [reply .gitmodules:18]: i have no clue, was looking at it too weirdly, quirk of git

**Review by flibidydibidy — APPROVED** [review]

**Comment by flibidydibidy** [comment]
Thank you @AviiNL and @Cook-I-T for getting continuous integration set up, this is going to save me a lot of time any annoyances around building the repository for windows users.

**Pull request merged**